app/src/main/java/cm/aptoide/pt/promotions/PromotionsFragment.java

```diff
@@ -37,16 +155,389 @@
+          .hasError()) {
+        handleDownloadError(promotionViewApp.getDownloadModel()
+            .getDownloadState());
+      }
+
+      walletActiveView.setVisibility(View.GONE);
+      walletInactiveView.setVisibility(View.VISIBLE);
+      ImageView appIcon = walletInactiveView.findViewById(R.id.app_icon);
+      TextView appName = walletInactiveView.findViewById(R.id.app_name);
+      TextView appDescription = walletInactiveView.findViewById(R.id.app_description);
+      TextView numberOfDownloads = walletInactiveView.findViewById(R.id.number_of_downloads);
+      TextView appSize = walletInactiveView.findViewById(R.id.app_size);
+      TextView rating = walletInactiveView.findViewById(R.id.rating);
+      Button promotionAction = walletInactiveView.findViewById(R.id.promotion_app_action_button);
+
+      ImageLoader.with(getContext())
+          .load(promotionViewApp.getAppIcon(), appIcon);
+      appName.setText(promotionViewApp.getName());
+      appDescription.setText(promotionViewApp.getDescription());
+      appSize.setText(AptoideUtils.StringU.formatBytes(promotionViewApp.getSize(), false));
+      if (promotionViewApp.getRating() == 0) {
+        rating.setText(R.string.appcardview_title_no_stars);
+      } else {
+        rating.setText(new DecimalFormat("0.0").format(promotionViewApp.getRating()));
+      }
+      numberOfDownloads.setText(String.valueOf(promotionViewApp.getNumberOfDownloads()));
+
+      promotionAction.setText(getContext().getString(getButtonMessage(getState(promotionViewApp)),
+          promotionViewApp.getAppcValue()));
+      if (getState(promotionViewApp) == CLAIMED) {
+        promotionAction.setEnabled(false);
+        promotionAction.setBackgroundColor(getContext().getResources()
+            .getColor(R.color.grey_fog_light));
+
+        promotionsAdapter.isWalletInstalled(true);
+      } else if (getState(promotionViewApp) == CLAIM) {
+        promotionAction.setEnabled(true);
+        promotionAction.setBackgroundColor(getContext().getResources()
+            .getColor(R.color.green));
+        promotionAction.setOnClickListener(__ -> promotionAppClick.onNext(
+            new PromotionAppClick(promotionViewApp, getClickType(getState(promotionViewApp)))));
+        promotionsAdapter.isWalletInstalled(true);
+      } else {
+        promotionAction.setEnabled(true);
+        TypedValue resultValue = new TypedValue();
+        getContext().getTheme()
+            .resolveAttribute(R.attr.installButtonBackground, resultValue, true);
+        if (resultValue.resourceId != 0) {
+          promotionAction.setBackgroundResource(resultValue.resourceId);
+        } else {
+          promotionAction.setBackgroundColor(getContext().getResources()
+              .getColor(R.color.orange));
+        }
+        promotionAction.setOnClickListener(__ -> promotionAppClick.onNext(
+            new PromotionAppClick(promotionViewApp, getClickType(getState(promotionViewApp)))));
+      }
+    }
+  }
```

Comment:
This should be extracted and reused because we are replicating this logic in each viewholder.